cmd/query/app/server_test.go

```diff
@@ -100,17 +100,61 @@ func TestServer(t *testing.T) {
 	assert.Equal(t, healthcheck.Unavailable, flagsSvc.HC().Get())
 }
 
+func TestServer2(t *testing.T) {
```

Comment:
This might deserve a better name. Can you describe what this particular is doing? What's different from it and the previous test? Suggestion: `TestServerWithDedicatedPorts`